src/mbgl/style/layers/symbol_layer_properties.hpp

```diff
@@ -31,9 +31,9 @@ struct IconIgnorePlacement : LayoutProperty<bool> {
     static bool defaultValue() { return false; }
 };
 
-struct IconImage : DataDrivenLayoutProperty<std::string> {
+struct IconImage : DataDrivenLayoutProperty<expression::Image> {
     static constexpr const char *name() { return "icon-image"; }
-    static std::string defaultValue() { return ""; }
+    static expression::Image defaultValue() { return ""; }
```

Comment:
nit: `return std::string(); // or {} to obviate conversion from C string to std::string, same in other places`

Reply:
fixed